src/services/challenges_repository.js

```diff
@@ -17,23 +16,35 @@ export default class ChallengesRepository {
 
   filterOutFinishedChallenges(allChallenges, resolvedChallenges, timedOutChallenges, ownAddress) {
     const timedOutSet = new Set(timedOutChallenges.map(({challengeId}) => challengeId));
-    const startedCount = allChallenges.reduce((acc, {challengeId, count}) => put(acc, challengeId, count), {});
+    const startedCount = allChallenges.reduce(
+      (acc, {challengeId, count}) => {
+        acc[challengeId] = count;
+        return acc;
+      },
+      {}
+    );
     const unresolvedCount = resolvedChallenges.reduce(
       (acc, {challengeId, resolverId}) => {
         if (resolverId === ownAddress) {
-          return put(acc, challengeId, 0);
+          acc[challengeId] = 0;
+        } else {
+          acc[challengeId] -= 1;
         }
-        return put(acc, challengeId, acc[challengeId] - 1);
+        return acc;
       },
       startedCount
     );
-    return allChallenges.filter(({challengeId}) => unresolvedCount[challengeId] > 0 && !timedOutSet.has(challengeId));
+    const filtered = allChallenges.filter(({challengeId}) => unresolvedCount[challengeId] > 0 && !timedOutSet.has(challengeId));
+    return filtered;
```

Comment:
I would inline `filtered`, but I presume it's just as with `sorted` elsewhere - done to make adding performance measurement easier.

Reply:
👍